frappe_whatsapp/utils/webhook.py

```diff
@@ -46,7 +46,7 @@ def post():
                     "message": message['text']['body']
                 }).insert(ignore_permissions=True)
     else:
-        update_status(data["entry"][0]["changes"][0])
+        update_status(data["entry"]["changes"][0])
```

Comment:
`{"object": "whatsapp_business_account", "entry": [{"id": "107455662167439", "changes": [{"value": {"messaging_product": "whatsapp", "metadata": {"display_phone_number": "15550483859", "phone_number_id": "109288118647518"}, "contacts": [{"profile": {"name": "Shridhar"}, "wa_id": "xxxxx"}], "messages": [{"context": {"from": "15550483859", "id": "wamid.HBgMOTE5NzQxMDk0NDY4FQIAERgSMzUxQjUwOUY3OTREODJCNjE1AA=="}, "from": "xxxxxx", "id": "wamid.HBgMOTE5NzQxMDk0NDY4FQIAEhgWM0VCMDQ5OEEzRDYxOTRGMjYyODJCMgA=", "timestamp": "1680094862", "type": "interactive", "interactive": {"type": "button_reply", "button_reply": {"id": "2", "title": "Second Button\u2019s Titl"}}}]}, "field": "messages"}]}], "cmd": "frappe_whatsapp.utils.webhook.webhook"}`

This is the data which we get in webhook entry[0] is correct. The change you have made will not capture the message

Reply:
In my case, I'm getting following response:
`{'object': 'whatsapp_business_account', 'entry': {'id': '103764272699039', 'changes': [{'value': {'messaging_product': 'whatsapp', 'metadata': {'display_phone_number': '15550256741', 'phone_number_id': '100188813063701'}, 'contacts': [{'profile': {'name': 'Vinay Rawat'}, 'wa_id': '919675705065'}], 'messages': [{'from': '919675705065', 'id': 'wamid.HBgMOTE5Njc1NzA1MDY1FQIAEhggNTIwQzBDQUJGMEI3NDAyNzcxQUEzMjZENzVBRkM3QzYA', 'timestamp': '1681798464', 'text': {'body': 'Raise an issue'}, 'type': 'text'}]}, 'field': 'messages'}]}, 'cmd': 'frappe_whatsapp.utils.webhook.webhook'}
`

Reply:
which version your using?

Reply:
Installed Apps
Call Log: v0.0.1 (vineet)
Core Erp: v0.0.1 (master)
ERPNext: v13.24.0 (version-13)
Frappe Framework: v13.24.0 (version-13)
Frappe Whatsapp: v0.0.2 (master)
Project: v0.0.1 (master)

Reply:
Whatsapp Webhook v16

Reply:
we need this to handle for both version of meta v15 and v16. Can you update the code to handle both cases

Reply:
Sure, will update you soon : )

Reply:
@shridarpatil Updated code for handling both v15 and v16